### ISSUE TITLE
Issue with ctrl latching while saving

### ISSUE THREAD

**Issue by takase1121** [drmargarido/linters#14]
Whenever I save a file, the linter runs and after the console window closes, the CTRL key is held down. I would need to press CTRL again to "unlatch" it.

**Comment by drmargarido** [comment]
Hello, thanks for the report.
To better understand the problem I need some extra context.
What you mean with "console window"? The lite console plugin, some terminal where you opened lite? Or other window?
What OS are you on?

**Comment by takase1121** [comment]
Sorry for not explaining throughly.

I am using Windows, and in Windows whenever you open a "console" program it will create a tty(? I call it a terminal) and that will pop up, taking the focus from the editor. That somehow messes with the keystroke, especially modifiers to "latch" until the next press is registered.

The console program I mentioned was the linter.

**Comment by takase1121** [comment]
Here is an example:
https://youtu.be/WA0lht-TOyg

**Comment by drmargarido** [comment]
Thanks for the video, now I understood the problem. I use archlinux and debian so I didn't see that happening and could not reproduce it.

I think a new way to call command line utilities without prompting the opening of the console on windows was introduced some versions ago in the lite core.

By preventing the opening of the console the focus of the window will not switch and the problem should be mitigated. I will check it and update the way linters call the tools.

**Comment by takase1121** [comment]
Thank you!

**Comment by takase1121** [comment]
I think the commit you referred to was this:
https://github.com/rxi/lite/commit/2b32edf7f0557959ed70d88fa0d31f275f560c36

This may or may not work for the purpose because we need the stdout/stderr, whereas this is just a `system` call

**Comment by drmargarido** [comment]
Leaving a comment just to say this issue is waiting for the answer of rxi to your PR to lite.

If it is accepted we can use it to implement this feature, otherwise we will need to think about another approach. The first simple possible thing that comes to my mind is to use exec, redirect stdout/stderr to a temporary file and then read the file. But for now let's wait.

**Comment by takase1121** [comment]
After _some_ waiting, I realised that my commit is un _`lite`_ ly to be merged (Sorry, I'll see myself out). I support your idea using `system.exec` and redirection to a temporary file. I am willing to help with this.

**Comment by takase1121** [comment]
Just to do a report on what I have done so far regarding the idea mentioned above:

I forked this repo and did some experiment with `os.execute`.
The idea is completely workable; I am no longer having this issue.
However, `lite` hangs when the linter is running because I used a while loop to "wait" for linter to complete.
This issue can be mitigated by using `core.add_thread` which I already have a dirty prototype with it that actually
worked flawlessly.
However the prototype still needs to be polished.
